sunkit_image/utils/noise.py

```diff
@@ -1,274 +1,277 @@
-"""
+'''
+This module implements a series of functions for noise level estimation
+'''
 
 import numpy as np
 from scipy.stats import gamma
 from skimage.util import view_as_windows
 from scipy.ndimage import correlate
 
-__all__ = ["NoiseLevelEstimation"]
-
-
-class NoiseLevelEstimation:
-    def __init__(self, img, patchsize=7, decim=0, conf=1 - 1e-6, itr=3):
-        """
-        Estimates the noise level of an image.
-
-        Additive white Gaussian noise (AWGN) is a basic noise model used in Information Theory
-        to mimic the effect of many random processes that occur in nature.
-
-        Parameters
-        ----------
-        img: `numpy.ndarray`
-            Single Numpy image array.
-        patchsize : `int`, optional
-            Patch size, defaults to 7.
-        decim : `int`, optional
-            Decimation factor, defaults to 0.
-            If you use large number, the calculation will be accelerated.
-        conf : `float`, optional
-            Confidence interval to determine the threshold for the weak texture.
-            In this algorithm, this value is usually set the value very close to one.
-            Defaults to 0.99.
-        itr : `int`, optional
-            Number of iterations,  defaults to 3.
-
-        Attributes
-        ----------
-        nlevel: `numpy.ndarray`
-            Estimated noise levels.
-        th: `numpy.ndarray`
-            Threshold to extract weak texture patches at the last iteration.
-        num: `numpy.ndarray`
-            Number of extracted weak texture patches at the last iteration.
-        mask: `numpy.ndarray`
-            Weak-texture mask.
-            0 and 1 represent non-weak-texture and weak-texture regions, respectively.
-
-        Examples
-        --------
-        >>> import numpy as np
-        >>> np.random.seed(0)
-        >>> noisy_image_array = np.random.randn(100, 100)
-        >>> estimate = NoiseLevelEstimation(noisy_image_array, patchsize=11, itr=10)
-        >>> estimate.mask
-        array([[1., 1., 1., ..., 1., 1., 0.],
-            [1., 1., 1., ..., 1., 1., 0.],
-            [1., 1., 1., ..., 1., 1., 0.],
-            ...,
-            [1., 1., 1., ..., 1., 1., 0.],
-            [1., 1., 1., ..., 1., 1., 0.],
-            [0., 0., 0., ..., 0., 0., 0.]])
-        >>> estimate.nlevel
-        array([1.0014616])
-        >>> estimate.th
-        array([173.61530607])
-        >>> estimate.num
-         array([8100.])
-
-        References
-        ----------
-        * Xinhao Liu, Masayuki Tanaka and Masatoshi Okutomi
-          Noise Level Estimation Using Weak Textured Patches of a Single Noisy Image
-          IEEE International Conference on Image Processing (ICIP), 2012.
-
-        * Xinhao Liu, Masayuki Tanaka and Masatoshi Okutomi
-          Single-Image Noise Level Estimation for Blind Denoising Noisy Image
-          IEEE Transactions on Image Processing, Vol.22, No.12, pp.5226-5237, December, 2013.
-        """
```

Comment:
You will want to keep most of this and put it into the new function.

Reply:
You're right, sorry. I wanted to avoid the examples bit as I haven't included working examples yet. I'll commit the references back into it and get some example code.

Reply:
Included references and updated example code in comment.